File: LeetCode_Marathons/LCM3/Hard/Minimum_Deletions_to_Make_Array_Divisible/a.java
class Solution {
    private int gcd(int number1, int number2) {
        while (number2 != 0) {
            int RemainderOfDivision = number1 % number2;
            number1 = number2;
            number2 = RemainderOfDivision;
        }
        return number1;
    }

    public int minOperations(int[] nums, int[] numsDivide) {
        int gcdOfNumsDivine = numsDivide[0];
        for (int number : numsDivide) {
            gcdOfNumsDivine = gcd(gcdOfNumsDivine, number);
        }
        Arrays.sort(nums);
        int counter = 0;
        for (int number : nums) {
            if (gcdOfNumsDivine % number == 0) {
                return counter;
            }
            counter++;
        }
        return -1;
    }
}

File: LeetCode_Marathons/LCM3/Hard/Minimum_Deletions_to_Make_Array_Divisible/a.py
class Solution:
    def gcd(number1: int, number2: int) -> int:
        while number2 != 0:
            number1, number2 = number2, number1 % number2
        return number1
    

    def minOperations(self, nums: List[int], numsDivide: List[int]) -> int:
        gcdOfNumsDivine: int = numsDivide[0]
        for number in numsDivide:
            gcdOfNumsDivine = gcd(gcdOfNumsDivine, number)
        nums.sort()
        counter: int = 0
        for number in nums:
            if gcdOfNumsDivine % number == 0: return counter
            counter += 1
        return -1